devito/types/sparse.py

```diff
@@ -72,11 +72,22 @@ def __shape_setup__(cls, **kwargs):
         if grid is None:
             raise TypeError('Need `grid` argument')
         shape = kwargs.get('shape')
+        dimensions = kwargs.get('dimensions')
         npoint = kwargs.get('npoint', kwargs.get('npoint_global'))
+        glb_npoint = SparseDistributor.decompose(npoint, grid.distributor)
         if shape is None:
-            glb_npoint = SparseDistributor.decompose(npoint, grid.distributor)
-            shape = (glb_npoint[grid.distributor.myrank],)
-        return shape
+            loc_shape = (glb_npoint[grid.distributor.myrank],)
+        else:
+            loc_shape = []
+            assert len(dimensions) == len(shape)
+            for i, (d, s) in enumerate(zip(dimensions, shape)):
+                if i is cls._sparse_position:
```

Comment:
intergers want == , though ininfluential here 

see here https://stackoverflow.com/questions/2239737/is-it-better-to-use-is-or-for-number-comparison-in-python#:~:text=Python%20uses%20the%20same%20object,this%20changes%20for%20bigger%20numbers.&text=So%20you%20should%20always%20use%20%3D%3D%20to%20compare%20numbers.&text=%3D%3D%20is%20what%20you%20want,to%20work%20on%20your%20examples.